pkg/autodiscovery/providers/file.go

```diff
@@ -150,6 +152,13 @@ func (c *FileConfigProvider) collectEntry(file os.FileInfo, path string, integra
 	entry := configEntry{}
 	absPath := filepath.Join(path, fileName)
 
+	// skip auto conf files based on the agent configuration
+	if fileName == "auto_conf.yaml" && containsString(config.Datadog.GetStringSlice("ignore_autoconf"), integrationName) {
```

Comment:
An extra safety would be to `strings.ToLower(fileName)` here.

Reply:
it's us who ship these files, they're not customers input 😅

Reply:
Then a const for the filename could have been great maybe? But a bit late to the party I guess.